src/email_service/auto_resume/lambda_function.py

```diff
@@ -0,0 +1,242 @@
+    return False
+
+
+def forward_message_to_target_queue(
+    message: dict[str, Any], target_queue_url: str
+) -> bool:
+    """
+    Forward a message from auto-resumer queue to a target queue.
+
+    :param message: The SQS message to forward
+    :return: True if message was forwarded successfully, False otherwise
+    """
+    try:
+        logger.info("Forwarding message to target SQS queue")
+        send_message_to_queue(target_queue_url, message)
+        logger.info("Successfully forwarded message to target SQS queue")
+        return True
+    except Exception as e:
+        logger.error("Failed to forward message to target SQS queue: %s", str(e))
+        return False
+
+
+def process_sqs_message(
+    message: dict[str, Any],
+    receipt_handle: str,
+    aws_request_id: str,
+    error_responder: ErrorResponder,
+) -> dict[str, Any]:
+    """
+    Process a single SQS message:
+    1. Ensure Aurora database is awake
+    2. Forward message to upsert_run queue
+    3. Delete message from auto-resumer queue
+
+    :param message: The parsed SQS message
+    :param receipt_handle: The receipt handle for the SQS message
+    :param aws_request_id: AWS request ID for tracking
+    :param target_sqs_url: The target SQS queue URL to forward the message to
+    :return: Response with status code and body
+    """
+    try:
+        # Try to wake up Aurora database
+        logger.info(
+            "Attempting to wake up Aurora database. Request ID: %s", aws_request_id
+        )
+
+        if not ensure_database_awake():
+            logger.error(
+                "Failed to wake up Aurora database after maximum retries. Request ID: %s",
+                aws_request_id,
+            )
+            # Don't delete the message, it will be retried automatically by SQS
+            return {
+                "statusCode": 500,
+                "body": json.dumps(
+                    {
+                        "message": "Failed to wake up Aurora database",
+                        "error": "Database connection error",
+                        "request_id": aws_request_id,
+                    }
+                ),
+            }
+
+        # Aurora is awake, forward message to upsert_run SQS queue
+        logger.info(
+            "Aurora database is awake, forwarding message to upsert_run SQS queue. Request ID: %s",
+            aws_request_id,
+        )
+
+        # Forward the message to upsert_run SQS queue
+        success_upsert = forward_message_to_target_queue(
+            message, UPSERT_RUN_SQS_QUEUE_URL
+        )
+        if not success_upsert:
+            return {
+                "statusCode": 500,
+                "body": json.dumps(
+                    {
+                        "message": "Failed to forward message to upsert_run SQS queue",
+                        "error": "Message forwarding error",
+                        "request_id": aws_request_id,
+                    }
+                ),
```

Comment:
Since auto-resumer is triggered by SQS rather than an HTTP request, returning HTTP-style responses (statusCode, body) isn’t meaningful. Instead, rely on raising exceptions for error handling. This ensures that failures propagate correctly and SQS–Lambda integration will retry messages as expected.

Ref: https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#sqs-polling-behavior